libgadget/winds.c

```diff
@@ -255,13 +261,48 @@ winds_and_feedback(int * NewStars, int NumNewStars, const double Time, const dou
     ta_free(priv->NPRedo);
     ta_free(priv->NPLeft);
 
+    /* Some particles may be kicked by multiple stars on the same timestep.
+     * To ensure this happens only once and does not depend on the order in
+     * which the loops are executed, particles are kicked by the nearest new star.*/
+    priv->StarKickVelocity = (double * ) mymalloc("NearestStar", SlotsManager->info[0].size * sizeof(double));
```

Comment:
What if two stars are of the same distance away from the particle?

Perhaps there is a more fundamental way to break the tie than tagging by the distance?

Reply:
I could break the tie with the particle ID?

My thinking with the distance was that this was modelling a shockwave from a supernova moving on sub-timestep scales, and the shockwave from the nearest star should arrive first.